animation_nodes/nodes/spline/splines_from_edges_utils.pyx

```diff
@@ -0,0 +1,117 @@
+from libc.string cimport memset
+from cpython.mem cimport PyMem_Malloc, PyMem_Realloc, PyMem_Free
+from ... data_structures cimport (
+    Mesh, Vector3DList, EdgeIndicesList, PolySpline,
+    VirtualDoubleList, FloatList
+)
+
+def splinesFromBranches(Vector3DList vertices, EdgeIndicesList edges, VirtualDoubleList radii):
+    cdef int i, j
+    cdef int edgesAmount = len(edges)
+    cdef int verticesAmount = len(vertices)
+
+    # Compute how many neighbour each vertex have.
+    cdef int *neighboursAmounts = <int*>PyMem_Malloc(verticesAmount * sizeof(int))
```

Comment:
in general it would be nice to use the Lists AN provides instead of doing memory allocations yourself if appropriate. The performance impact should be negilible.